src/oscar/templates/oscar/catalogue/detail.html

```diff
@@ -9,11 +9,11 @@
 {% load purchase_info_tags %}
 
 {% block title %}
-    {{ product.get_title }} | {{ block.super }}
+    {{ product.meta_title|default:product.get_title|default:""|striptags }} | {{ block.super }}
 {% endblock %}
 
 {% block description %}
-    {{ product.description|default:""|striptags }}
+    {{ product.meta_description|default:product.description|default:""|striptags }}
```

Comment:
As above, please define a `get_meta_description` method to handle obtaining the correct meta description.